pom.xml

```diff
@@ -1,117 +1,54 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-jar-plugin</artifactId>
-                <!-- According to the Maven wiki, we have to use at least 3.2 for reproducible builds -->
-                <!-- https://maven.apache.org/guides/mini/guide-reproducible-builds.html -->
-                <version>3.2.0</version>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.8.1</version>
+                <configuration>
+                    <source>${java.version}</source>
+                    <target>${java.version}</target>
+                </configuration>
             </plugin>
-
-            <!-- Add libraries to the plugin -->
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-shade-plugin</artifactId>
                 <version>3.2.4</version>
-                <configuration>
-                    <createDependencyReducedPom>false</createDependencyReducedPom>
-                    <relocations>
-                        <relocation>
-                            <pattern>oshi</pattern>
-                            <shadedPattern>lagmonitor.oshi</shadedPattern>
-                            <excludes>
-                                <exclude>*.properties</exclude>
-                                <exclude>oshi.architecture.properties</exclude>
-                                <exclude>oshi.linux.filename.properties</exclude>
-                                <exclude>oshi.macos.versions.properties</exclude>
-                                <exclude>oshi.properties</exclude>
-                                <exclude>oshi.vmmacaddr.properties</exclude>
-                            </excludes>
-                        </relocation>
-                        <relocation>
-                            <pattern>org.slf4j</pattern>
-                            <shadedPattern>lagmonitor.slf4j</shadedPattern>
-                        </relocation>
-                    </relocations>
-                    <artifactSet>
-                        <excludes>
-                            <!-- Exclude native drivers by default for user security reasons -->
-                            <exclude>net.java.dev.jna:jna</exclude>
-                        </excludes>
-                    </artifactSet>
-                </configuration>
                 <executions>
                     <execution>
                         <phase>package</phase>
                         <goals>
                             <goal>shade</goal>
                         </goals>
+                        <configuration>
+                            <createDependencyReducedPom>false</createDependencyReducedPom>
+                        </configuration>
                     </execution>
                 </executions>
             </plugin>
-
-            <!--Expose git variables for version names-->
```

Comment:
This necessary to versioning the plugin.yml file. It exposes a variable that is used there.